protocol-documentation.md

```diff
@@ -0,0 +1,122 @@
+# Protocol documentation
+Early version of the protocol used for communication, subject to be changed and expanded. Not all messages are implemented.
+
+# Format of messages
+
+Every message that is sent between the client and the server must be in the format described below. The `type` field describes what type of request/response is being sent, and the `payload` value contains whatever data is associated with that message. Important to note here is that payload can have many different types, i.e it can be an object, array or a specific value. Please look at the specification for each individual message to know what data the payload contains.
+
+```json
+{
+    "type": "<unique description of request/response>",
+    "payload": {
+        "data": []
+    }
+}
+```
+
+# Client messages
+Messages that are sent from a client to the server.
+
+When the position or zoom on the client's map is changed, this message should be sent to the server so that only information about busses that can be seen are sent to the client.
+```json
+{
+    "type": "geo-position-update",
+    "payload": {
+        "radius": 10,
+        "position": {
+            "latitude": 59,
+            "longitude": 16
+        }
+    }
+}
+```
+
+Sent to get information about a specific line.
+> Not implemented
+```json
+{
+    "type": "get-line-info",
+    "payload": {
+        "line": 5
+    }
+}
+```
+
+Sent to reserve a seat on a bus with a specific id.
+> Not implemented
+```json
+{
+    "type": "reserve-seat",
+    "payload": {
+        "id": 123456
```

Comment:
Does the Client map GTFS ids to UL bus lines?

Reply:
We're still looking in to how this will work exactly. `reserve-seat` and some other messages were added with a "Not implemented" text above them to indicate that we *will* implement them later.